apis/core/lib/domain/csv-source/upload.ts

```diff
@@ -59,6 +59,20 @@ export async function uploadFile({
           .addOut(csvw.header, true)
           .addOut(csvw.headerRowCount, header.length)
       })
+
+    for (let index = 0; index < header.length; index++) {
+      const columnName = header[index]
+      csvSource.addOut(csvw.column, id.column(csvSource, columnName), column => {
+        column.addOut(rdf.type, [csvw.Column])
+          .addOut(schema.name, columnName)
+          .addOut(dtype.order, index)
+        rows.forEach(row => {
+          if (row.length > index) {
+            column.addOut(cc.csvColumnSample, row[index])
```

Comment:
This is good enough for a start, but it would be great to have a sample of multiple unique values of each column.

Reply:
Was thinking about this as well. We might also have to reiterate over the extract sample code. Id columns might screw this up.